sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -840,12 +854,15 @@ pi_result _pi_queue::resetCommandList(pi_command_list_ptr_t CommandList,
           ? this->Context->ZeCopyCommandListCache[this->Device->ZeDevice]
           : this->Context->ZeComputeCommandListCache[this->Device->ZeDevice];
 
-  // Fence had been signalled meaning the associated command-list completed.
-  // Reset the fence and put the command list into a cache for reuse in PI
-  // calls.
-  ZE_CALL(zeFenceReset, (CommandList->second.ZeFence));
-  ZE_CALL(zeCommandListReset, (CommandList->first));
-  CommandList->second.InUse = false;
+  // Immediate commandlists do not have an associated fence.
+  if (!UseImmediateCommandLists) {
```

Comment:
I thought we agreed that you'd just check if fence != nullptr

Reply:
> The function resetCommandList never needs to be called for immediate commandlists.
I have added checks to ensure that, and then, the body of the resetCommandList function can stay as before.

If it not called, then maybe we don't need this change at all? (but add an assert that fence is not null)

Reply:
That was around line 3309, but it can be done here too.

Reply:
Same is for other checks in this function, let's check for the command-list properties rather than some global setting

Reply:
OK